src/vdbfusion/vdbfusion/VDBVolume.cpp

```diff
@@ -143,9 +160,21 @@ void VDBVolume::Integrate(const std::vector<Eigen::Vector3d>& points,
                 const float new_tsdf = (last_tsdf * last_weight + tsdf * weight) / (new_weight);
                 tsdf_acc.setValue(voxel, new_tsdf);
                 weights_acc.setValue(voxel, new_weight);
+                if (has_colors) {
```

Comment:
I always wanted to extract these fusion functions outside the main loop, but I guess this is too much refactoring to ask. Once this PR is merged, I'll attempt to fix this design and ask for your review if you are willing to do it ;)

Reply:
There is an `VDBVolume:UpdateTSDF` function which is currently not really used but seems to do the same things. Is this a performance issue?

I can move it to use that function and define something similar for colors (or integrate it into that function), if it is not a performance issue. There are the two additional calls to:

```
AccessorRW tsdf_acc = AccessorRW(tsdf_->tree());
AccessorRW weights_acc = AccessorRW(weights_->tree());
```

But I suspect that is not much overhead.